### PR TITLE
Fix waypoint layer position bug by adding ability to use custom parent layer position

### LINKED ISSUE #3887: [Bug]: Label layers are above the waypoint layers.
### Mapbox Navigation SDK version

v2.5.0-beta.3 

### Steps to reproduce

1. Build the `Example` app.
2. Choose multiple waypoints, place the waypoints on the road label
3. Start navigation

### Expected behavior

The waypoint layer is above the road labels.

### Actual behavior

The waypoint layer is below the label layers as below. Checking the layer position, the waypoint layer is below all the label layers and road shield layers.
waypointLayer

### Is this a one-time issue or a repeatable issue?

repeatable

### PR DESCRIPTION
**Pull request by jill-cardamon**
### Description

This PR fixes #3887 by adding the ability to use a custom waypoint parent layer position.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -1159,8 +1163,10 @@ open class NavigationMapView: UIView {
                     let circlesLayer = delegate?.navigationMapView(self,
                                                                    waypointCircleLayerWithIdentifier: waypointCircleLayerIdentifier,
                                                                    sourceIdentifier: waypointSourceIdentifier) ?? defaultWaypointCircleLayer()
-                    
-                    if mapView.mapboxMap.style.layerExists(withId: NavigationMapView.LayerIdentifier.arrowSymbolLayer) {
+                    // Use custom layer position for the waypoint layer.
+                    if (layerPosition != nil) {
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView.swift:1167]: Can you cover these three conditions with unit tests? It'd be important to have them to make sure that we did not break anything.

Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView.swift:1167]:
```suggestion
                    if let layerPosition = layerPosition {
```

File: Sources/MapboxNavigation/NavigationMapView.swift [hunk 2]
```diff
@@ -1125,9 +1125,13 @@ open class NavigationMapView: UIView {
      Adds the route waypoints to the map given the current leg index. Previous waypoints for completed legs will be omitted.
      
      - parameter route: `Route`, on which a certain `Waypoint` will be shown.
+     - parameter layerPosition: Position of the waypoint layer. Defaults to `nil`.
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView.swift:1128]: Please also add changelog entry.

File: Tests/MapboxNavigationTests/NavigationMapViewTests.swift [hunk 3]
```diff
@@ -782,4 +782,78 @@ class NavigationMapViewTests: TestCase {
...
+        let customFixture = Fixture.route(waypoints: coordinates)
+        
+        navigationMapView.show([customFixture.route])
+        navigationMapView.showWaypoints(on: customFixture.route)
+        
+        // Style doesn't contain any layers besides the waypoint layer, main route layer, and the main route casing layer.
+        // If layer position wasn't provided, waypoint later should be placed
+        // above the main route layer.
+        // Since the waypoint layer identifier is generated by `showWaypoints(on:layerPosition:legIndex:)`,
+        // we must verify by layer type.
```
Comment (MaximAlien) [inline Tests/MapboxNavigationTests/NavigationMapViewTests.swift:827]: Waypoint layers will always have `MapboxNavigation-MapboxNavigation-resources_waypointCircleLayer` or `MapboxNavigation-MapboxNavigation-resources_waypointSymbolLayer` identifiers. I think you can use `NavigationMapView.LayerIdentifier.waypointCircleLayer` and `NavigationMapView.LayerIdentifier.waypointSymbolLayer` constants in tests as well.

**Comment by MaximAlien** [comment]
@jill-cardamon, can you please rebase this PR and check whether it aligns well with https://github.com/mapbox/mapbox-navigation-ios/pull/4030?

**Comment by jill-cardamon** [comment]
It doesn't align with the new flow. We should address custom layering in separate PR.

**Pull request closed without merging**